x-pack/plugins/apm/public/hooks/use_apm_breadcrumbs.ts

```diff
@@ -206,10 +191,6 @@ export function useBreadcrumbs(routes: APMRouteDefinition[]) {
     wrappedGetAPMHref,
     navigateToUrl,
   });
-  const title = getTitleFromBreadcrumbs(breadcrumbs);
 
-  useEffect(() => {
-    changeTitle(title);
-    setBreadcrumbs(breadcrumbs);
-  }, [breadcrumbs, changeTitle, location, title, setBreadcrumbs]);
+  useBreadcrumbs(breadcrumbs);
```

Comment:
I see we still have `routeDefinitionsToBreadcrumbs` and `addLinksToBreadcrumbs`. I know those two are necessary in the current implementation but can we make them redundant somehow?
How are other plugins doing this?

Reply:
APM has more pages than other observability apps. We can define the breadcrumbs on each page, but there's going to be a lot of repetition. I think other plugins are specifying the whole list on each page.

What we're doing is working and well-tested and makes it so we only need to define the breadcrumb text once for each route. `useApmBreadcrumbs` figures out how to construct the list and the links based on the route definitions.